posthog/api/organization.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Dict
+
+from django.db import transaction
+from django.db.models import QuerySet, query
+from django.shortcuts import get_object_or_404
+from rest_framework import (
+    exceptions,
+    mixins,
+    permissions,
+    request,
+    response,
+    serializers,
+    status,
+    viewsets,
+)
+from rest_framework.generics import DestroyAPIView
+from rest_framework.mixins import DestroyModelMixin
+
+from posthog.models import Organization, OrganizationMembership, Team, team
+from posthog.permissions import CREATE_METHODS, OrganizationAdminWritePermissions, OrganizationMemberPermissions
+
+
+class PremiumMultiorganizationPermissions(permissions.BasePermission):
```

Comment:
It seems like an actual permissions thing to me (especially since 402 is very rarely used) plus permission classes are very nice to use. Not sure if this is doable in a similarly simple way without permission classes.